tethysext/atcore/tests/integrated_tests/models/files_database/file_collection_client_tests.py

```diff
@@ -463,3 +474,327 @@ def test_bad_collection_client(self):
+
+    def test_duplicate_item_directory_does_not_exist(self):
+        """Test duplicating and item that does not exist throws the correct error."""
+        test_dir_name = 'test_duplicate_item_directory_does_not_exist'
+        base_files_root_dir = os.path.join(self.test_files_base, test_dir_name)
+        root_dir = os.path.join(self.test_files_base, 'temp', test_dir_name)
+        self.copy_files_to_temp_directory(base_files_root_dir, root_dir)
+        _, collection_instance = self.get_database_and_collection(
+            database_id=self.general_database_id, collection_id=self.general_collection_id,
+            root_directory=root_dir, database_meta={}, collection_meta={}
+        )
+        collection_client = FileCollectionClient(self.session, self.general_collection_id)
+        with self.assertRaises(FileCollectionItemNotFoundError) as exc:
+            collection_client.duplicate_item('dir2', 'duplicated_dir2')
+        self.assertTrue('"dir2" not found in this collection.' in str(exc.exception))
+        self.assertFalse(os.path.exists(os.path.join(collection_client.path, 'duplicated_dir2')))
+
+    def test_duplicate_item_already_exists(self):
+        """Test duplicating and item to a target that exists throws the correct error."""
+        test_dir_name = 'test_duplicate_item_already_exists'
+        base_files_root_dir = os.path.join(self.test_files_base, test_dir_name)
+        root_dir = os.path.join(self.test_files_base, 'temp', test_dir_name)
+        self.copy_files_to_temp_directory(base_files_root_dir, root_dir)
+        _, collection_instance = self.get_database_and_collection(
+            database_id=self.general_database_id, collection_id=self.general_collection_id,
+            root_directory=root_dir, database_meta={}, collection_meta={}
+        )
+        collection_client = FileCollectionClient(self.session, self.general_collection_id)
+        with self.assertRaises(FileCollectionItemAlreadyExistsError) as exc:
+            collection_client.duplicate_item('file1.txt', 'duplicated_file1.txt')
+        self.assertTrue('Collection duplication target already exists.' in str(exc.exception))
+
+    def test_delete_item(self):
+        """Test deleting an item."""
+        test_dir_name = 'test_delete_item'
+        base_files_root_dir = os.path.join(self.test_files_base, test_dir_name)
+        root_dir = os.path.join(self.test_files_base, 'temp', test_dir_name)
+        self.copy_files_to_temp_directory(base_files_root_dir, root_dir)
+        _, collection_instance = self.get_database_and_collection(
+            database_id=self.general_database_id, collection_id=self.general_collection_id,
+            root_directory=root_dir, database_meta={}, collection_meta={}
+        )
+        collection_client = FileCollectionClient(self.session, self.general_collection_id)
+        collection_client.delete_item('file1.txt')
+        self.assertFalse(os.path.exists(os.path.join(collection_client.path, 'file1.txt')))
```

Comment:
Assert that it exists before calling delete_item

Reply:
done